### ISSUE TITLE
Overridden theme colors returns unexpected background color

### ISSUE THREAD

**Issue by filoxo** [jxnblk/mdx-deck#195]
Hey there! I just started using mdx and generated a new deck. I began tinkering with the theme and stumbled upon some unexpected/interesting behavior. Here are the steps to reproduce:

1. Create new deck, `npm init deck my-deck`
2. Edit the generated *theme.js* file
3. Add colors key with an object that contains a text key with a valid color string
    - Note that the default theme is already imported

```js
import theme from 'mdx-deck/themes'

export default {
  ...theme,
  // Customize your presentation theme here.
  // custom colors
  colors: {
    text: '#333', 
  }
}
```

### Expected output

The server reloads the page with only the text color changed. 

### Actual output

The server reloads the page with the text color changed and now includes a purple-blue color as the background, despite it not having been modified or included in the base theme. 

![screen shot 2018-10-09 at 4 28 49 pm](https://user-images.githubusercontent.com/4202993/46702480-791f3000-cbe0-11e8-8440-543449a474f3.png)

I also tested this with other themes, and am seeing the same behavior. 

#### Workaround
Of course its easy enough to simply declare the background again. 

```js
  colors: {
    text: '#333',
    background: '#fff',
  }
```

I'd be interested in poking around and seeing if I can find where this is happening, but I'm not sure if I'll have a lot of time. I'd be happy to contribute if I could get pointed in the right direction. Thanks!

**Comment by filoxo** [comment]
Another workaround. Using a deep merge (I used lodash's `merge` utility here) would give you the theme defaults. 

```javascript
import theme from 'mdx-deck/themes'
import merge from 'lodash.merge'

const customTheme = merge({...theme}, {
  // custom colors
  colors: {
    text: '#333',
  }
})

export default customTheme
```

Using a shallow merge methodology (like `Object.assign`) still produces the same issue, which is expected because colors is a nested object and Object.assign does not recursively merge.

```javascript
import theme from 'mdx-deck/themes'

const customTheme = Object.assign({...theme}, {
  // custom colors
  colors: {
    text: '#333',
  }
})

// still produces blue background color
export default customTheme
```

**Comment by jxnblk** [comment]
You should include a defined `background` color in your theme – the color you're seeing is the browser default for the `background` keyword which is being used if an explicit `background` isn't set in your theme

**Comment by filoxo** [comment]
I did a little diving and while I still can't explain the reason for the exact color (see [Why does html think chucknorris is a color](https://stackoverflow.com/questions/8318911/why-does-html-think-chucknorris-is-a-color) and [randomstringtocsscolor](http://randomstringtocsscolor.com/) for potential fun) I do think there is still a bug here. `background` is not a valid CSS color or keyword, and does not display uniformly across browsers. I don't think that the output CSS should include such values if the user has not provided a color for this property. I noticed a similar incorrect value with the H1 that is being rendered. It has `color: heading;` which is correctly being invalidated by Chrome. 
However, this could be happening due to styled-components and not mdx.

**Comment by jxnblk** [comment]
`Background` is a system color (not recommended for use) https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#System_Colors

The docs could be clarified, but **you need to include a `colors` object for the theming to work correctly** – for the current version, this is expected behavior

**Comment by filoxo** [comment]
Thanks, I understand this better. Based on your statement, I found the following code interesting  (found in the Theming docs):

```js
export default {
  // extends the default theme
  ...theme,
  // add a custom font
  font: 'Roboto, sans-serif',
  // custom colors
  colors: {
    text: '#f0f',
    background: 'black',
    link: '#0ff',
  }
}
```
because this in effect does not return a fully valid `colors` object (as noted in [theming#reference](theming.md#reference)). To do as you're saying, us as consumers would have to use a deep merge strategy (as I noted above) or manually merge the inner `colors` object

```js
export default {
  // extends the default theme
  ...theme,
  // add a custom font
  font: 'Roboto, sans-serif',
  // custom colors
  colors: {
    ...theme.colors, // Include all colors
    text: '#f0f',
    background: 'black',
    link: '#0ff',
  }
}
```

The docs are therefore encouraging use of code that may produce this unexpected styling behavior. That's how I stumbled across this. I only wanted to change  the text and link color of the dark theme but retain the rest, and using the above code sample caused the behavior first identified in this issue. 

For the time being I'll submit a PR to clarify the themes doc about this potential pitfall. I think in an ideal world though, mdx themes shouldn't return "not recommended" style values.
